src/Service.GraphQLBuilder/Mutations/CreateMutationBuilder.cs

```diff
@@ -221,47 +272,49 @@ private static InputValueDefinitionNode GenerateScalarInputType(NameNode name, F
                 defaultValue = value.Fields[0].Value;
             }
 
+            bool isFieldNullable = defaultValue is not null ||
+                (!IsNoSQLDb(databaseType) && DoesRelationalDBSupportNestedCreate(databaseType) && DoesFieldHaveReferencingFieldDirective(fieldDefinition));
+
             return new(
                 location: null,
                 fieldDefinition.Name,
                 new StringValueNode($"Input for field {fieldDefinition.Name} on type {GenerateInputTypeName(name.Value)}"),
-                defaultValue is not null ||
-                (DoesRelationalDBSupportNestedMutations(databaseType) && DoesFieldHaveReferencingFieldDirective(fieldDefinition)) ? fieldDefinition.Type.NullableType() : fieldDefinition.Type,
+                isFieldNullable ? fieldDefinition.Type.NullableType() : fieldDefinition.Type,
                 defaultValue,
                 new List<DirectiveNode>()
             );
         }
 
         /// <summary>
-        /// Generates a GraphQL Input Type value for an object type, generally one provided from the database.
+        /// Generates a GraphQL Input Type value for an object type, generally one provided from the relational database.
```

Comment:
The sentence already says object type. And yes, it can be used to create input types for source->target linking object as well. I'll add it to the comment.